include/nuttx/video/fb.h

```diff
@@ -232,258 +232,296 @@
+#define FB_ACTIVATE_MASK             15    /* values                         */
+#define FB_ACTIVATE_VBL              16    /* Activate values on next vbl    */
+#define FB_CHANGE_CMAP_VBL           32    /* Change colormap on vbl         */
+#define FB_ACTIVATE_ALL              64    /* Change all VCs on this fb      */
+#define FB_ACTIVATE_FORCE            128   /* Force apply even if no change  */
+#define FB_ACTIVATE_INV_MODE         256   /* Invalidate videomode           */
+
+#define FB_ACCELF_TEXT               1     /* (OBSOLETE) See fb_info.flags   */
+                                           /* and vc_mode                    */
+
+#define FB_SYNC_HOR_HIGH_ACT         1     /* Horizontal sync high active    */
+#define FB_SYNC_VERT_HIGH_ACT        2     /* Vertical sync high active      */
+#define FB_SYNC_EXT                  4     /* External sync                  */
+#define FB_SYNC_COMP_HIGH_ACT        8     /* Composite sync high active     */
+#define FB_SYNC_BROADCAST            16    /* Broadcast video timings        */
+                                           /* vtotal = 144d/288n/576i -  PAL */
+                                           /* vtotal = 121d/242n/484i - NTSC */
+#define FB_SYNC_ON_GREEN             32    /* Sync on green                  */
+
+#define FB_VMODE_NONINTERLACED       0     /* Non interlaced */
+#define FB_VMODE_INTERLACED          1     /* Interlaced */
+#define FB_VMODE_DOUBLE              2     /* Double scan */
+#define FB_VMODE_ODD_FLD_FIRST       4     /* Interlaced: top line first */
 #define FB_VMODE_MASK                255
 
-#define FB_VMODE_YWRAP               256    /* Ywrap instead of panning */
-#define FB_VMODE_SMOOTH_XPAN         512    /* Smooth xpan possible */
-                                            /* (internally used) */
-#define FB_VMODE_CONUPDATE           512    /* Don't update x/yoffset */
+#define FB_VMODE_YWRAP               256   /* Ywrap instead of panning */
+#define FB_VMODE_SMOOTH_XPAN         512   /* Smooth xpan possible */
+                                           /* (internally used) */
+#define FB_VMODE_CONUPDATE           512   /* Don't update x/yoffset */
 
 #define FB_ROTATE_UR                 0
 #define FB_ROTATE_CW                 1
 #define FB_ROTATE_UD                 2
 #define FB_ROTATE_CCW                3
 
+#define ARGBTO32(a,r,g,b) \
+  ((uint32_t)((a) & 0xff) << 24 | (uint32_t)((r) & 0xff) << 16 | \
+   (uint32_t)((g) & 0xff) << 8  | (uint32_t)((b) & 0xff))
+
+#define RGBTO24(r,g,b) \
+  ((uint32_t)((r) & 0xff) << 16 | (uint32_t)((g) & 0xff) << 8 | \
+   (uint32_t)((b) & 0xff))
+
+#define RGBTO16(r,g,b) \
+  ((((uint16_t)(r) << 8) & 0xf800) | (((uint16_t)(g) << 3) & 0x07e0) | \
+   (((uint16_t)(b) >> 3) & 0x001f))
+
+#define RGBTO8(r,g,b) \
+  ((((uint8_t)(r) << 5) & 0xe0) | (((uint8_t)(g) << 2) & 0x1c) | \
+  ((uint8_t)(b) & 0x03))
+
+#ifdef CONFIG_VIDEO_FB_SPLASHSCREEN
```

Comment:
move to source file

Reply:
It's used in the logo .c files so fb.h is included in those. But I will double check where it needs to be included and move if appropriate.

Reply:
Format conversions are back in rgbcolors.h where they belong.